### PR TITLE
Fix footgun advice re loading polyfills

### LINKED ISSUE #2787: Fix footgun advice re polyfills
See https://webpack.js.org/guides/shimming/#loading-polyfills 
> including polyfills in the main bundle is not recommended

This is very bad advice. Even modern browsers potentially need every polyfill to repair their broken implementations; the best practice is to unconditionally and synchronously load all needed polyfills.

Specifically, it's exceedingly hard to safely asynchronously load polyfills, because to be safe, polyfills must run before *all other code* on the page ("first-run code" is the only kind of robustness available to JS).

I agree that you may want to be smarter about which polyfills you send where, but the whole point is that people make those decisions incorrectly - I believe latest Chrome even needs polyfills for some things, and a year or three ago, there were **zero** browsers that fully supported ES5. The bundle size cost is a necessary evil, and is much less problematic than incorrectness.

(posted per @TheLarkInn)

Can we update this to, at the least, not recommend a dangerous practice by incorrectly prioritizing bundle size over correctness?

### PR DESCRIPTION
**Pull request by ljharb**
Fixes #2787.

### CODE REVIEW

File: src/content/guides/shimming.md [hunk 1]
```diff
@@ -285,8 +285,10 @@ __src/index.js__
 
 T> Note that we aren't binding the `import` to a variable. This is because polyfills simply run on their own, prior to the rest of the code base, allowing us to then assume certain native functionality exists.
 
-Now while this is one approach, __including polyfills in the main bundle is not recommended__ because this penalizes modern browsers users by making them download a bigger file with unneeded scripts.
+Note that this approach prioritizes correctness over bundle size. To be safe and robust, polyfills/shims must run __before all other code__, and thus either need to load synchronously, or, all app code needs to load after all polyfills/shims load. There are many misconceptions in the community, as well, that modern browsers "don't need" polyfills, or that polyfills/shims merely serve to add missing features - in fact, they often *repair broken implementations*, even in the most modern of browsers. The best practice thus remains to unconditionally and synchronously load all polyfills/shims, despite the bundle size cost this incurs.
```
Comment (ljharb) [inline src/content/guides/shimming.md]:
```suggestion
Note that this approach prioritizes correctness over bundle size. To be safe and robust, polyfills/shims must run __before all other code__, and thus either need to load synchronously, or, all app code needs to load after all polyfills/shims load. There are many misconceptions in the community, as well, that modern browsers "don't need" polyfills, or that polyfills/shims merely serve to add missing features - in fact, they often _repair broken implementations_, even in the most modern of browsers. The best practice thus remains to unconditionally and synchronously load all polyfills/shims, despite the bundle size cost this incurs.
```

File: src/content/guides/shimming.md [hunk 2]
```diff
@@ -285,8 +285,10 @@ __src/index.js__
 
 T> Note that we aren't binding the `import` to a variable. This is because polyfills simply run on their own, prior to the rest of the code base, allowing us to then assume certain native functionality exists.
 
-Now while this is one approach, __including polyfills in the main bundle is not recommended__ because this penalizes modern browsers users by making them download a bigger file with unneeded scripts.
+Note that this approach prioritizes correctness over bundle size. To be safe and robust, polyfills/shims must run __before all other code__, and thus either need to load synchronously, or, all app code needs to load after all polyfills/shims load. There are many misconceptions in the community, as well, that modern browsers "don't need" polyfills, or that polyfills/shims merely serve to add missing features - in fact, they often _repair broken implementations_, even in the most modern of browsers. The best practice thus remains to unconditionally and synchronously load all polyfills/shims, despite the bundle size cost this incurs.
```
Comment (ljharb) [inline src/content/guides/shimming.md]:
```suggestion
Note that this approach prioritizes correctness over bundle size. To be safe and robust, polyfills/shims must run __before all other code__, and thus either need to load synchronously, or, all app code needs to load after all polyfills/shims load.
There are many misconceptions in the community, as well, that modern browsers "don't need" polyfills, or that polyfills/shims merely serve to add missing features - in fact, they often _repair broken implementations_, even in the most modern of browsers.
The best practice thus remains to unconditionally and synchronously load all polyfills/shims, despite the bundle size cost this incurs.
```

File: src/content/guides/shimming.md [hunk 3]
```diff
@@ -285,8 +285,12 @@ __src/index.js__
 
 T> Note that we aren't binding the `import` to a variable. This is because polyfills simply run on their own, prior to the rest of the code base, allowing us to then assume certain native functionality exists.
 
-Now while this is one approach, __including polyfills in the main bundle is not recommended__ because this penalizes modern browsers users by making them download a bigger file with unneeded scripts.
+Note that this approach prioritizes correctness over bundle size. To be safe and robust, polyfills/shims must run __before all other code__, and thus either need to load synchronously, or, all app code needs to load after all polyfills/shims load.
+There are many misconceptions in the community, as well, that modern browsers "don't need" polyfills, or that polyfills/shims merely serve to add missing features - in fact, they often _repair broken implementations_, even in the most modern of browsers.
+The best practice thus remains to unconditionally and synchronously load all polyfills/shims, despite the bundle size cost this incurs.
 
+If you feel that you have mitigated these concerns and wish to incur the risk of brokenness, here's one way you might do it:
+d
```
Comment (montogeek) [inline src/content/guides/shimming.md]:
```suggestion
```

Just remove the `d`

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**